### PR TITLE
Pix3D TFRecord Generator for Mesh R-CNN

### PR DESCRIPTION
**Pull request by davidliii**
# Description

This PR adds a TFRecord Generator and Dataloader for Pix3D dataset using Mesh R-CNN annotations. The annotations and were obtained from the official Mesh R-CNN repository: https://github.com/facebookresearch/meshrcnn/blob/main/datasets/pix3d/download_pix3d.sh

In addition to the COCO annotations, the Pix3D TFRecord Generator produces the following new annotations:

| Name | Shape | Dtype | Description |
| - | - | - | - |
| model/vertices | [None, 3] | float32 | The (x,y,z) coordinates of the mesh vertices |
| model/faces | [None, 3] | int64 | The indices of the 3 vertices composing each face |
| model/voxel_indices | [None, 3] | int64 | The indices of the occupied voxels in the voxel grid |
| model/voxel_shape | [3] | int64 | The shape of the voxel grid |
| model/rot_mat | [3, 3] | float32 | Matrix to rotate the mesh coordinates so that it is aligned it to the image |
| model/trans_mat | [3] | float32 | Vector to translate the mesh coordinates so that it is aligned it to the image |
| model/intrinsic_mat | [3] | float32 | Vector containing the intrinsic camera parameters: f, px, py |

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] A new research paper code implementation

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * TFRecord Generator Test:  `python -m official.vision.beta.projects.mesh_rcnn.data.create_pix3d_tf_record_test`
> * Dataloader Test: `python -m official.vision.beta.projects.mesh_rcnn.dataloaders.meshrcnn_input_test`
> * Preprocess Ops Test: `python -m official.vision.beta.projects.mesh_rcnn.ops.meshrcnn_preprocess_ops_test`

**Test Configuration**:
In `create_pix3d_tf_record_test`, ensure that `pix3d_dir` is set to the directory where the raw Pix3D dataset is located. A test annotation file is provided under official/vision/beta/projects/mesh_rcnn/data called `tfrecord_pix3d_test_annotations.json`. This file should be copied into the raw dataset directory.

In `meshrcnn_input_test`, alter the `input_path` string in the DataConfig class to the path of the TFRecord directory. This string should correspond to the `OUTPUT_DIR/FILE_PREFIX` flag when running `create_pix3d_tf_record`.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/vision/beta/projects/mesh_rcnn/data/create_pix3d_tf_record.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+def create_tf_example(image):
+  """Converts image and annotations to a tf.Example proto.
+  Args:
+    image: dict with keys:
+      ['filename, 'height', 'width', 'iscrowd', 'segmentation', 'model',
+          'category_id', 'K', 'bbox', 'trans_mat', 'area', 'image_id',
+          'rot_mat', 'voxel', 'pix3d_dir']
+  Returns:
+    example: The converted tf.Example
```
Comment (vishnubanna) [inline official/vision/beta/projects/mesh_rcnn/data/create_pix3d_tf_record.py:135]: add punctuation.

File: official/vision/beta/projects/mesh_rcnn/data/pix3d_visualizer.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (vishnubanna) [inline official/vision/beta/projects/mesh_rcnn/data/pix3d_visualizer.py:7]: remove the visulization code from the PR

File: official/vision/beta/projects/mesh_rcnn/data/create_pix3d_tf_record.py (hunk 1 above)
Comment (davidliii) [reply official/vision/beta/projects/mesh_rcnn/data/create_pix3d_tf_record.py]: docstrings have been updated

File: official/vision/beta/projects/mesh_rcnn/dataloaders/meshrcnn_input_test.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+        self.assertAllEqual(
+            tf.shape(gt_faces_mask),
+            [batch_size, params.parser.max_num_faces])
+
+        self.assertAllEqual(tf.shape(rot_mat), [batch_size, 3, 3])
+        self.assertAllEqual(tf.shape(trans_mat), [batch_size, 3])
+        self.assertAllEqual(tf.shape(trans_mat), [batch_size, 3])
+
+        if i >= 5:
```
Comment (davisjam) [inline official/vision/beta/projects/mesh_rcnn/dataloaders/meshrcnn_input_test.py:191]: Possibly cleaner to change the for loop to use a slice of dataset? `for (image, labels) in dataset[0:6]` or so?

Comment (davidliii) [reply official/vision/beta/projects/mesh_rcnn/dataloaders/meshrcnn_input_test.py:191]: The dataset object is not subscriptable.
